stock_available_mrp/tests/test_potential_qty.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+# © 2014 Numérigraphe SARL
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.tests.common import TransactionCase
+from openerp.osv.expression import TRUE_LEAF
+
+
+class TestPotentialQty(TransactionCase):
+    """Test the potential quantity on a product with a multi-line BoM"""
+
+    def setUp(self):
+        super(TestPotentialQty, self).setUp()
+
+        #  An interesting product (multi-line BoM, variants)
+        self.tmpl = self.browse_ref(
+            'product.product_product_4_product_template')
+        #  First variant
+        self.var1 = self.browse_ref('product.product_product_4c')
+        #  Second variant
+        self.var2 = self.browse_ref('product.product_product_4')
+        # Components that can be used to make the product
+        component_ids = [
+            # CPUa8
+            self.ref('product.product_product_23'),
+            # RAM-SR2
+            self.ref('product.product_product_14'),
+            # HDD SH-2 replaces RAM-SR2 through our demo phantom BoM
+            self.ref('product.product_product_18'),
+            # RAM-SR3
+            self.ref('product.product_product_15')]
+
+        # Zero-out the inventory of all variants and components
+        for component_id in (
+                component_ids + [v.id
+                                 for v in self.tmpl.product_variant_ids]):
+            inventory = self.env['stock.inventory'].create(
+                {'name': 'no components: %s' % component_id,
+                 'location_id': self.ref('stock.stock_location_locations'),
+                 'filter': 'product',
+                 'product_id': component_id,
+                 })
+            inventory.prepare_inventory()
+            inventory.reset_real_qty()
+            inventory.action_done()
+
+        #  A product without a BoM
+        self.product_wo_bom = self.browse_ref('product.product_product_23')
+
+        #  Record the initial quantity available for sale
+        self.initial_usable_qties = {i.id: i.immediately_usable_qty
+                                     for i in [self.tmpl,
+                                               self.var1,
+                                               self.var2,
+                                               self.product_wo_bom]}
+
+        # Get the warehouses
+        self.wh_main = self.browse_ref('stock.warehouse0')
+        self.wh_ch = self.browse_ref('stock.stock_warehouse_shop0')
+
+    def assertPotentialQty(self, record, qty, msg):
+        record.refresh()
+        #  Check the potential
+        self.assertEqual(record.potential_qty, qty, msg)
```

Comment:
Travis seems to fail here. @lmignon could it be related to the change in multi-company rules at https://github.com/odoo/odoo/commit/2fd14db57433d08ab368a7e18d18ae8d49dafab1 by any chance?